oak_functions/loader/src/server.rs

```diff
@@ -153,15 +153,26 @@ pub trait ExtensionFactory {
 /// `Uwabi` extension called by listening to a channel.
 pub enum BoxedExtension {
     Native(Box<dyn OakApiNativeExtension + Send + Sync>),
-    Uwabi(Box<dyn UwabiExtension + Send + Sync>),
+    Uwabi(BoxedUwabiExtension),
 }
 
+pub type BoxedUwabiExtension = Box<dyn UwabiExtension + Send + Sync>;
+
 pub type BoxedExtensionFactory = Box<dyn ExtensionFactory + Send + Sync>;
 
 /// Trait for implementing an extension which relies on UWABI.
 pub trait UwabiExtension {
     /// Get the channel handle to address this extension.
     fn get_channel_handle(&self) -> ChannelHandle;
+
+    /// Get the endpoint.
+    fn get_endpoint(&mut self) -> Option<&mut Endpoint>;
```

Comment:
Maybe rename to `get_endpoint_mut`, usually that's explicit when returning mut refs.